rhtconsulting/rhc-ose
# Cleaned up hostname role to make it more generic

#### What does this PR do?

Cleans up the hostname role to avoid using loops to set hostname. Also, removes the DNS tie to make the hostnames role more generic. 
#### How should this be manually tested?

Provision an environment (e.g.: in OS1) and observe how the hostnames stays the same, but allows for custom dns domain - i.e.: set in the inventory file.

```
./provision.sh -i=<path_to_inventory> -p=<path_to_OSE_installer>
```

Also, custom hostnames can be set with the `custom_hostname` variable. To best test this, one may want to create a new playbook that just tests the `hostnames` role.
#### Is there a relevant Issue open for this?

resolves #180 
resolves #181 
resolves #183 
#### Who would you like to review this?

/cc @etsauer @JayKayy
